### PR TITLE
Include DOI link icon in the anchor.

### PR DESCRIPTION
**Pull request by EreMaijala**
I believe the icon should be part of the link. We're currently quite inconsistent [1] with that, but the results view type links can be used as an example. It causes the unsightly underlining issue, but still better for usability and the icon getting the link color.

[1] e.g. `Show QR Code` and `Save to List` don't include the icon, while `Add to Book Bag` does.

### CODE REVIEW

**Comment by demiankatz** [comment]
@EreMaijala, #2400 is work in progress to normalize all the link/icon behavior. @crhallberg just finished another round of bug-fixing there, so I'm planning to test it again this morning. Maybe we should hold off on finishing this until that one is merged.

**Comment by EreMaijala** [comment]
@demiankatz #2400 is not affecting this one, but I'm fine with incorporating the fixes there if it seems to be nearing completion.

**Comment by demiankatz** [comment]
@EreMaijala, if you don't expect #2400 to cause a need for further changes here, I'm happy to review/merge this now... I just thought if the styling there would be useful for improving this code, then we might as well wait instead of changing the code twice.

**Comment by EreMaijala** [comment]
@demiankatz Further changes are needed here to get the underlining nice and uniform like in #2400. But since #2400 is pretty extensive, I can't merge it to our tree without fair amount of work, so my wish would be to merge this (then it could be cherry-picked or merged before #2400 lands), then merge dev to #2400 and add the tweaks to doi.js there. But if it's too much trouble, I can just override doi.js locally and undo that later.

**Comment by demiankatz** [comment]
@EreMaijala, if that makes your life easier, it's fine with me -- I just want to be sure we don't forget about this, but that's easily done by adding a TODO checkbox on #2400 as soon as this is merged. I'll try to take a closer look here as soon as I'm done with my current #2400 review.

**Comment by EreMaijala** [comment]
@demiankatz I've already added a TODO checkbox in #2400 since it needs attention in any case.

**Review by demiankatz — APPROVED** [review]
Looks fine to me!

**Pull request merged**
